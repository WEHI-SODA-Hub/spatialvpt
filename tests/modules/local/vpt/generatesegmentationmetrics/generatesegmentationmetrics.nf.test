nextflow_process {

    name "Test Process VPT_GENERATESEGMENTATIONMETRICS"
    script "modules/local/vpt/generatesegmentationmetrics/generatesegmentationmetrics.nf"
    process "VPT_GENERATESEGMENTATIONMETRICS"

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id: 'small_set' ],
                    [ file('${projectDir}/tests/data/cell_by_gene_repartitioned.csv', checkIfExists: true) ],
                    [ file('${projectDir}/tests/data/cell_metadata_resegmented.csv', checkIfExists: true) ],
                    [ file('${projectDir}/data/202305010900_U2OS_small_set_VMSC00000/region_0/detected_transcripts.csv', checkIfExists: true) ],
                    [ file('${projectDir}/data/202305010900_U2OS_small_set_VMSC00000/region_0/images', checkIfExists: true) ],
                    [ file('${projectDir}/tests/data/cellpose_micron_space.parquet', checkIfExists: true) ],
                    [ file('${projectDir}/data/202305010900_U2OS_small_set_VMSC00000/region_0/images/micron_to_mosaic_pixel_transform.csv', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
