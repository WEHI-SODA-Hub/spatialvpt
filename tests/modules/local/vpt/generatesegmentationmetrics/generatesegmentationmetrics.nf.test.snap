{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "small_set"
                        },
                        "small_set_metrics.html:md5,7815f1201383a7a5bce15efcd24bd5db"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "small_set"
                        },
                        "small_set_metrics.csv:md5,5cef8ebfe90c8771fc19ca23dfe130c4"
                    ]
                ],
                "2": [
                    "versions.yml:md5,85bfd4bf0802aa541a2be6278b69fbb4"
                ],
                "metrics": [
                    [
                        {
                            "id": "small_set"
                        },
                        "small_set_metrics.csv:md5,5cef8ebfe90c8771fc19ca23dfe130c4"
                    ]
                ],
                "report": [
                    [
                        {
                            "id": "small_set"
                        },
                        "small_set_metrics.html:md5,7815f1201383a7a5bce15efcd24bd5db"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,85bfd4bf0802aa541a2be6278b69fbb4"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-08-26T11:51:22.368525681"
    }
}